client/rpc/src/lib.rs

```diff
@@ -329,20 +329,18 @@ where
         Ok(SaveFileToDisk::Success(file_metadata))
     }
 
-    async fn get_forest_root(&self, forest_key: Option<H256>) -> RpcResult<H256> {
+    async fn get_forest_root(&self, forest_key: Option<H256>) -> RpcResult<Option<H256>> {
         let forest_key = FSH::Key::from(forest_key.unwrap_or_default().as_ref().to_vec());
 
-        let fs = self
-            .forest_storage_handler
-            .get(&forest_key)
-            .await
-            .ok_or_else(|| {
-                into_rpc_error(format!("Forest storage not found for key {:?}", forest_key))
-            })?;
+        // return None if not found
+        let fs = match self.forest_storage_handler.get(&forest_key).await {
+            Some(fs) => fs,
+            None => return Ok(None),
```

Comment:
Why returning `None` instead of an error?

Reply:
This is analogous to a substrate storage element with `OptionQuery`. I return `None` if there was no element found under that key. An error should be thrown when querying storage failed for some reason other than the requested key not existing.